x-pack/plugins/observability/server/services/slo/aggregations/get_timeslice_metric_indicator_aggregation.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { TimesliceMetricIndicator, timesliceMetricMetricDef } from '@kbn/slo-schema';
+import * as t from 'io-ts';
+
+import { getElastichsearchQueryOrThrow } from '../transform_generators';
+
+type TimesliceMetricDef = TimesliceMetricIndicator['params']['metric'];
+type TimesliceMetricMetricDef = t.TypeOf<typeof timesliceMetricMetricDef>;
+
+export class GetTimesliceMetricIndicatorAggregation {
+  constructor(private indicator: TimesliceMetricIndicator) {}
+
+  private buildAggregation(metric: TimesliceMetricMetricDef) {
+    switch (metric.aggregation) {
```

Comment:
If the type permits it (like we do for the SLI type), we should handle all aggregation cases specifically and use the `default` as a type check with `assertNever(aggregation)`. 
The only reason I see for not doing this is if we are sure no more aggregation will be added in the future.

Reply:
Done